npm/generate_katas_content.js

```diff
@@ -64,19 +84,74 @@ function tryReadFile(filePath, errorPrefix) {
   try {
     content = readFileSync(filePath, "utf8");
   } catch (e) {
-    throw new Error(`${errorPrefix}: ${e}`);
+    throw new Error(`${errorPrefix}\n${e}`);
   }
   return content;
 }
 
-function getMissingProperties(properties, required) {
+function identifyMissingProperties(properties, required) {
   return required.filter((property) => !Object.hasOwn(properties, property));
 }
 
-function generateExampleSection(kataPath, properties) {
+function getSourceId(sourcePath, basePath) {
+  return relative(basePath, sourcePath).replaceAll(sep, "__");
+}
+
+function aggregateSources(paths, globalCodeSources) {
+  const codeSources = [];
+  for (const path of paths) {
+    const id = getSourceId(path, globalCodeSources.basePath);
+    if (!(id in globalCodeSources.sources)) {
+      const code = tryReadFile(path, "Could not read code dependency");
+      globalCodeSources.sources[id] = code;
+    }
+    codeSources.push(id);
+  }
+  return codeSources;
+}
+
+function parseMarkdown(markdown) {
+  const tokens = [];
+  const macroRegex = /@\[(?<type>\w+)\]\((?<json>\{.*?\})\)\n/gs;
```

Comment:
You may want to make the newline detection be (`\r?\n``), just in case someone has checked out with Windows line endings.

Reply:
Updated.